### ISSUE TITLE
Pillow doesn't open PNG with color profile

### ISSUE THREAD

**Issue by JohnCC330** [python-pillow/Pillow#5672]
This program:

![logo_moodle](https://user-images.githubusercontent.com/3728987/129263844-3fb37023-aa50-4da8-aa96-71ea30f48ce5.png)

```python
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
from PIL import Image
import os.path as osp

IMG = osp.expanduser("logo_moodle.png")

def main(args):
    print(Image.__version__)
    img = Image.open(IMG)
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

```
Fails open with error: 

`PIL.UnidentifiedImageError: cannot identify image file '/root/doc/!univ/cg/logo_moodle.png'
`

Removing color profile (with GIMP), solves the problem. There was a rather large block
with XML at the end ('CCIPCC'?)

* OS:  Linux Slackware-current
* Python: 3.8.5
* Pillow: 8.3.1

**Comment by radarhere** [comment]
The error behind `UnidentifedImageError` is `SyntaxError: broken PNG file (bad header checksum in b'iCCP')`

If I run [pngcheck](http://www.libpng.org/pub/png/apps/pngcheck.html) (`pngcheck logo_moodle.png`), I get
```
logo_moodle.png  CRC error in chunk iCCP (computed cba88e56, expected f2af8eee)
ERROR: logo_moodle.png
```

So it would seem that your image is corrupted.

To workaround this, you can ask Pillow to load truncated images. The following code opens the image successfully.
```python
from PIL import Image, ImageFile
ImageFile.LOAD_TRUNCATED_IMAGES = True
import os.path as osp

IMG = osp.expanduser("logo_moodle.png")

def main(args):
    print(Image.__version__)
    img = Image.open(IMG)
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
```

**Comment by radarhere** [comment]
@JohnCC330 has that solved your problem, or did you have further thoughts?

**Comment by JohnCC330** [comment]
Yes, Andrew - it explained why the error was there. But I'm hesitant about disabling validity checks.

I compiled pngcheck (3.0.3) and it gave the error you mentioned. I was considering to submit a bug
report to GIMP as it reloads the image without detecting the CRC issue itself. I then stumbled upon
[a bug report there which documents the issue](https://gitlab.gnome.org/GNOME/gimp/-/issues/2111).
The issue was raised two years ago, and as far as I can see, still hasn't been solved.

One of the comments there confirms that disabling saving the Color Profile (in GIMP) works around the
problem and looks more correct than blindfolding PIL.

Cheers!
